cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/Utils.java

```diff
@@ -0,0 +1,89 @@
+package zone.cogni.libs.sparqlservice.impl;
+
+import java.text.MessageFormat;
+import java.util.Objects;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.PasswordAuthentication;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+/**
+ * Utility methods for JDK11 HttpClient.
+ */
+public class Utils {
+
+  private static final Logger log = LoggerFactory.getLogger(Utils.class);
+
+  /**
+   * Creates a basic authenticator (username, password)
+   *
+   * @param username username
+   * @param password password (not null)
+   * @return authenticator
+   */
+  public static Authenticator createBasicAuthenticator(final String username,
+      final String password) {
+    Objects.requireNonNull(password);
+    return new Authenticator() {
+      @Override
+      protected PasswordAuthentication getPasswordAuthentication() {
+        return new PasswordAuthentication(username, password.toCharArray());
+      }
+    };
+  }
+
+  /**
+   * Creates a new HttpClientBuilder. If both username and password is supplied, basic authentication header is generated.
+   *
+   * @param username username
+   * @param password password
+   * @return HttpClient.Builder
+   */
+  public static HttpClient.Builder createHttpClientBuilder(final String username,
+      final String password) {
+    final HttpClient.Builder httpClientBuilder = HttpClient.newBuilder();
+
+    if (StringUtils.isNoneBlank(username, password)) {
+      httpClientBuilder.authenticator(createBasicAuthenticator(username, password));
+    } else {
+      log.error("Endpoint credentials not properly configured");
```

Comment:
a little soft maybe. depends on what is checked before `createHttpClientBuilder` is called also.

Reply:
This logic tries to unify different behaviour for username/password that was exhibited by the different backend implementations before (different conditions on username/password being (not)null). Currently:
- both username and password present means authentication should be used,
- either username and password not present means authentication should not be used.
In this sense, it is quite simple.

An alternative:
- both username/password is present -> authentication should be used,
- none of username/password is present -> authentication should not be used,
- either username/password is not present -> an exception is thrown.